src/libmongoc/tests/test-mongoc-bulk.c

```diff
@@ -4679,6 +4679,240 @@ test_bulk_bypass_document_validation (void)
+      r = mongoc_bulk_operation_replace_one_with_opts (
+         bulk,
+         tmp_bson ("{'$expr': {'$eq': ['$_id', '$$id']}}"),
+         tmp_bson ("{'x': 'foo'}"),
+         NULL,
+         &error);
+      expect_msg = "{'$db': 'test',"
+                   " 'update': 'test',"
+                   " 'let': {'id': 1},"
+                   " 'ordered': true}";
+      expect_doc = "{'q': {'$expr': {'$eq': ['$_id', '$$id']}},"
+                   " 'u': {'x': 'foo'},"
+                   " 'multi': false}";
+      break;
+   case BULK_UPDATE:
+      r = mongoc_bulk_operation_update_many_with_opts (
+         bulk,
+         tmp_bson ("{'$expr': {'$eq': ['$_id', '$$id']}}"),
+         tmp_bson ("{'$set': {'x': 'foo'}}"),
+         NULL,
+         &error);
+      expect_msg = "{'$db': 'test',"
+                   " 'update': 'test',"
+                   " 'let': {'id': 1},"
+                   " 'ordered': true}";
+      expect_doc = "{'q': {'$expr': {'$eq': ['$_id', '$$id']}},"
+                   " 'u': {'$set': {'x': 'foo'}},"
+                   " 'multi': true}";
+      break;
+   case BULK_UPDATE_ONE:
+      r = mongoc_bulk_operation_update_one_with_opts (
+         bulk,
+         tmp_bson ("{'$expr': {'$eq': ['$_id', '$$id']}}"),
+         tmp_bson ("{'$set': {'x': 'foo'}}"),
+         NULL,
+         &error);
+      expect_msg = "{'$db': 'test',"
+                   " 'update': 'test',"
+                   " 'let': {'id': 1},"
+                   " 'ordered': true}";
+      expect_doc = "{'q': {'$expr': {'$eq': ['$_id', '$$id']}},"
+                   " 'u': {'$set': {'x': 'foo'}},"
+                   " 'multi': false}";
+      break;
+   default:
+      BSON_ASSERT (false);
+   }
+
+   ASSERT_OR_PRINT (r, error);
+   future = future_bulk_operation_execute (bulk, &reply, &error);
+
+   request = mock_server_receives_msg (mock_server,
+                                       MONGOC_MSG_NONE,
+                                       tmp_bson (expect_msg),
+                                       tmp_bson (expect_doc));
+
+   mock_server_replies_simple (request, "{'ok': 1.0, 'n': 1}");
+   request_destroy (request);
+   ASSERT (future_get_uint32_t (future));
+   future_destroy (future);
+
+   bson_destroy (&reply);
+   mongoc_bulk_operation_destroy (bulk);
+   mongoc_collection_destroy (collection);
+   mongoc_client_destroy (client);
+   mock_server_destroy (mock_server);
+}
+
+
+static void
+test_bulk_let (void)
+{
+   _test_bulk_let (BULK_REMOVE);
+   _test_bulk_let (BULK_REMOVE_ONE);
+   _test_bulk_let (BULK_REPLACE_ONE);
+   _test_bulk_let (BULK_UPDATE);
+   _test_bulk_let (BULK_UPDATE_ONE);
+}
+
+
+static void
+test_bulk_let_multi (void)
+{
+   mock_server_t *mock_server;
+   mongoc_client_t *client;
+   mongoc_collection_t *collection;
+   mongoc_bulk_operation_t *bulk;
+   bson_t reply;
+   bson_error_t error;
+   request_t *request;
+   future_t *future;
+   const char *expect_msg;
+   const char *expect_doc;
+   bool r;
+
```

Comment:
```suggestion

```

Please remove unused declarations. It causes the [debug-compile-with-warnings](https://spruce.mongodb.com/task/mongo_c_driver_releng_debug_compile_with_warnings_patch_6764dc21060386b6d89cf8f508ebfbf2631f0764_6230d4f13066154f97f76fc7_22_03_15_18_03_30/logs?execution=0) task to fail.